timeutil/schedule.go

```diff
@@ -82,40 +91,267 @@ func ParseTime(s string) (t TimeOfDay, err error) {
 	return t, nil
 }
 
-// Schedule defines a start and end time in which events should run.
-type Schedule struct {
-	Start TimeOfDay
-	End   TimeOfDay
+// Week represents a weekday such as Monday, Tuesday, with optional
+// week-in-the-month position, eg. the first Monday of the month
+type Week struct {
+	Weekday time.Weekday
+	// Pos defines which week inside the month the Day refers to, where zero
+	// means every week, 1 means first occurrence of the weekday, and 5
+	// means last occurrence (which might be the fourth or the fifth).
+	Pos uint
```

Comment:
Nitpick:

```
typedef WeekOfMonth uint
const (
  EveryWeek WeekOfMonth = iota,
  FirstWeekOfMonth,
  SecondWeekOfMonth,
  ThirdWeekOfMonth,
  FourthWeekOfMonth,
  LastWeekOfMonth,
)
```

Reply:
IIRC we've gone back and fort a bit on this. I can try and add:
```
const (
   EveryWeek uint = 0
   LastWeek uint = 5
)
```

Reply:
What we have now seems better. The rationale is that the numbers is what people will see (except for zero, which is the default). It's awkward to map the number which is publicly visible into type names. It does the opposite of what a constant name is generally supposed to do. mon2 is monday, the second. We don't need a different name for 2.